addons/website_quote/models/order.py

```diff
@@ -1,190 +1,166 @@
 # -*- coding: utf-8 -*-
-##############################################################################
-#
-#    OpenERP, Open Source Management Solution
-#    Copyright (C) 2013-Today OpenERP SA (<http://www.openerp.com>).
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
-
-from openerp.osv import osv, fields
+
 import uuid
 import time
 import datetime
 
+from openerp import api, fields, models
+
 import openerp.addons.decimal_precision as dp
 
-class sale_quote_template(osv.osv):
+
+class sale_quote_template(models.Model):
     _name = "sale.quote.template"
     _description = "Sale Quotation Template"
-    _columns = {
-        'name': fields.char('Quotation Template', required=True),
-        'website_description': fields.html('Description', translate=True),
-        'quote_line': fields.one2many('sale.quote.line', 'quote_id', 'Quotation Template Lines', copy=True),
-        'note': fields.text('Terms and conditions'),
-        'options': fields.one2many('sale.quote.option', 'template_id', 'Optional Products Lines', copy=True),
-        'number_of_days': fields.integer('Quotation Duration', help='Number of days for the validity date computation of the quotation'),
-    }
-    def open_template(self, cr, uid, quote_id, context=None):
+
+    name = fields.Char('Quotation Template', required=True)
+    website_description = fields.Html('Description', translate=True)
+    quote_line = fields.One2many('sale.quote.line', 'quote_id', string='Quotation Template Lines', copy=True)
+    note = fields.Text('Terms and conditions')
+    options = fields.One2many('sale.quote.option', 'template_id', string='Optional Products Lines', copy=True)
+    number_of_days = fields.Integer(string='Quotation Duration', help='Number of days for the validity date computation of the quotation')
+
+    @api.multi
+    def open_template(self):
```

Comment:
yes its called by type of object button. use @api.multi with self.ensure_one().then you can use self.id